### PR TITLE
Update CUReviews API Route

### LINKED ISSUE #885: Bottom Bar Displays "N/A" for Most Fields
The bottom bar is currently broken, and does not display any information fetched from CUReviews' API; see the following screenshot:

Screenshot 2024-02-12 at 1 48 49 AM

This is due to the routes of CUReviews' API [changing](https://github.com/cornell-dti/course-reviews-react-2.0/pull/424/commits/c15a70f4a2152ea136a88becc10cf5b3892dcc4c).

### PR DESCRIPTION
**Pull request by zachary-kent**
### Summary 

Updates a reference to CUReviews' API; see [here](https://github.com/cornell-dti/course-reviews-react-2.0/pull/424/commits/c15a70f4a2152ea136a88becc10cf5b3892dcc4c). We should merge this and release ASAP, as the bottom bar is currently broken in prod. Closes #885

It also bumps the node version in our snapshot action so that the checks don't fail. Unfortunately, I have to make both of these changes in the same PR so we can merge. Gross, I know.

### Test Plan 

CI

### CODE REVIEW

**Comment by zachary-kent** [comment]
Argh, I'm cursed with "Expected — Waiting for status to be reported" @elizabeth-tang you had this issue before right? Do you remember how you resolved it?

File: src/components/BottomBar/BottomBarState.ts
Comment (Destaq) [inline src/components/BottomBar/BottomBarState.ts:1]: Beautifully executed. Clean and efficient. You should be proud.

File: .github/workflows/cd-snapshot.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
       - name: Set up Node
         uses: actions/setup-node@v2
         with:
-          node-version: '16'
+          node-version: '20'
```
Comment (Destaq) [inline .github/workflows/cd-snapshot.yml:12]: Not sure if this has anything to do with the WIP issue (which seems to be resolved now). I checked but there were no GitHub incidents about Actions today / yesterday, wonder what could have happened. In any case seems like functionality is back up after what ET tried?

Comment (zachary-kent) [reply .github/workflows/cd-snapshot.yml:12]: No this is to fix the snapshot action failing. It's failing on every other open PR too.

**Review by andxu282 — APPROVED** [review]
LGTM if we can fix the WIP

**Pull request closed without merging**